python/hail/expr/functions.py

```diff
@@ -1778,7 +1778,159 @@ def rand_unif(min, max, seed=None) -> Float64Expression:
+        (Optional) Left boundary of truncated beta distribution.
+    max : float or :class:`.Expression` of type :py:data:`.tfloat64`
+        (Optional) Right boundary of truncated beta distribution.
+    seed : :obj:`int` or `None`
+        If not `None`, function will be seeded with provided seed.
+
+    Returns
+    -------
+    :class:`.Expression` of type :py:data:`.tfloat64`
+    """
+    if min is None and max is None:
+        return _seeded_func("rand_beta", tfloat64, seed, a, b)
+    if min is None:
+        min = hl.literal(0)
+    if max is None:
+        max = hl.literal(1)
+
+    return _seeded_func("rand_beta", tfloat64, seed, a, b, min, max)
+
+
+@typecheck(a=expr_float64,
+           scale=expr_float64,
+           seed=nullable(int))
+def rand_gamma(a, scale, seed=None) -> Float64Expression:
+    """Returns a random floating-point number drawn from a gamma distribution
+    with parameters `a` and `scale`.
+
+    Notes
+    -----
+    The optional parameters `min` and `max` represent a truncated beta
+    distribution with parameters a and b and support `[min, max]`. Draws are
+    made via rejection sampling, i.e. returning the first draw from Beta(a,b)
+    that falls in range `[min, max]`. This procedure may be slow if the
+    probability mass of Beta(a,b) over `[min, max]` is small.
+
+    Examples
+    --------
+
+    >>> hl.rand_gamma(1, 1).value
+    2.3915947710237537
+
+    >>> hl.rand_gamma(1, 1).value
+    0.1339939936379711
+
+    Parameters
+    ----------
+    a : float or :class:`.Expression` of type :py:data:`.tfloat64`
+    scale : float or :class:`.Expression` of type :py:data:`.tfloat64`
+    seed : :obj:`int` or `None`
+        If not `None`, function will be seeded with provided seed.
+
+    Returns
+    -------
+    :class:`.Expression` of type :py:data:`.tfloat64`
+    """
+    return _seeded_func("rand_gamma", tfloat64, seed, a, scale)
+
+
+@typecheck(probabilities=expr_array(expr_float64),
+           seed=nullable(int))
+def rand_draw(probabilities, seed=None) -> Int32Expression:
+    """Returns a random index from an array of unnormalized probabilities.
+
+    Warning
+    -------
+    This function may be slow over very large arrays.
+
+    Examples
+    --------
+
+    >>> hl.rand_draw([0, 1.7, 2]).value
+    2
+
+    >>> hl.rand_draw([0, 1.7, 2]).value
+    1
+
+    Parameters
+    ----------
+    probabilities : float or :class:`.Expression` of type :py:data:`.tfloat64`
+    seed : :obj:`int` or `None`
+        If not `None`, function will be seeded with provided seed.
+
+    Returns
+    -------
+    :class:`.Expression` of type :py:data:`.tfloat64`
+    """
+    return _seeded_func("rand_draw", tint32, seed, probabilities)
+
+
+@typecheck(a=expr_array(expr_float64),
+           seed=nullable(int))
+def rand_dirichlet(a, seed=None) -> ArrayExpression:
+    """Returns a sample from the Dirichlet distribution parameterized by vector
```

Comment:
"Returns a random draw from a Dirichlet distribution." with link to 
https://en.wikipedia.org/wiki/Dirichlet_distribution